testsuite/features/step_definitions/common_steps.rb

```diff
@@ -1332,11 +1332,11 @@ def server_secret
 end
 
 Then(/^port "([^"]*)" should be (open|closed)$/) do |port, selection|
-  output, _code = $server.run("lsof -i -P -n  | grep #{port}  || echo 'closed'", check_errors: false)
-  case selection
-  when 'open'
-    raise "Port '#{port}' not open although it should be!" if output == "closed\n"
-  when 'closed'
-    raise "Port '#{port}' open although it should not be!" if output != "closed\n"
+  _output, code = $server.run("ss --listening --numeric | grep :#{port}", check_errors: false, verbose: true)
```

Comment:
is there any reason why we need the `:` before the port being grepped?

Reply:
The intention is to avoid catching false positive port 13333 when grepping 3333.
Even more bulletproof would be:
`ss --listening --numeric srv :#{port} | grep #{port}`

Reply:
`ss --listening --numeric srv :#{port} | grep #{port}` does not work but `ss --listening --numeric -H src :#{port} | grep #{port}` I guess?

```bash
head-srv:~ # ss --listening --numeric srv :5556 | grep 5556
Error: an inet prefix is expected rather than "srv".
Cannot parse dst/src address.

head-srv:~ # ss --listening --numeric -H src :5556 | grep 5556
tcp LISTEN 0      3      *:5556 *:*
```
But yes, the semicolon is intentional.